packages/AsyncLoader/src/AsyncLoader.js

```diff
@@ -0,0 +1,134 @@
+            this,
+            canReadAttributes([{
+                name: 'data-endpoint-url',
+                property: 'endpointURL',
+                validate: (value) => !!value,
+            }, {
+                name: 'data-trigger-event-name',
+                property: 'triggerEventName',
+                validate: (value) => !!value,
+            }, {
+                name: 'data-trigger-event-filter',
+                property: 'triggerEventFilter',
+            }]),
+        );
+        this.readAttributes();
+    }
+
+    connectedCallback() {
+        this.#setupTriggerEventListener();
+    }
+
+    disconnectedCallback() {
+        this.#teardownTriggerEventListener();
+    }
+
+    /**
+     * Listen to event specified in data-trigger-event-name
+     */
+    #setupTriggerEventListener() {
+        this.#teardownTriggerEventListener = createListener(
+            window,
+            this.triggerEventName,
+            this.#handleTiggerEvent.bind(this),
+        );
+    }
+
+    /**
+     * Tests if event dispatched passes filter in case trigger-event-filter was provided
+     */
+    #handleTiggerEvent(event) {
+        if (this.triggerEventFilter) {
+            // Do not use eval to limit scope of variables that can be accessed; still use JS in the
+            // attribute to allow for maximum flexibility.
+            try {
+                const filterFunction = Function('event', `return ${this.triggerEventFilter}`);
+                const eventMatchesFilter = filterFunction(event);
+                if (!eventMatchesFilter) return;
+            } catch (error) {
+                // Make sure the user understands where the error came from
+                error.message = `The filter function provided through the trigger-event-filter attribute of <async-loader> threw the following error: ${error.message}`;
+                throw error;
+            }
+        }
+        this.#fetchData();
+    }
+
+    async #fetchData() {
+        this.#displayTemplate('[data-loading-template]');
+        try {
+            const response = await fetch(this.endpointURL);
+            if (!response.ok) {
+                this.#displayError(`Status ${response.status}`);
+            } else {
+                const content = await response.text();
+                this.#getContentContainer().innerHTML = content;
+            }
+        } catch (error) {
+            this.#displayError(error.message);
+            // Do not prevent error from being handled correctly; JSDOM displays an "Unhandled
+            // rejection" error, therefore ignore it for now
+            // throw error;
+        }
+    }
+
+    #getContentContainer() {
+        const container = this.querySelector('[data-content-container]');
+        if (!container) {
+            throw new Error('AsyncLoader: Could not find container to place content within; no child element matches selector [data-content-container].');
+        }
+        return container;
+    }
```

Comment:
Willst du das nicht in einer Class-Property speichern? Dann wird es nur einmal aufgerufen anstatt 2-3x.

Reply:
Ich hatte mir das überlegt, ja. Überlegung hinter dieser Lösung: Das Innenleben der Komponente könnte man von aussen verändern, z.B. via JavaScript. Dann wäre plötzlich ein anderer Container der `[data-content-container]`. Deswegen mach ich's zur Laufzeit, um sicher das richtige Element zu erwischen. Wenn ich's bei `connectedCallback` speichern würde, wär's dann vielleicht gar nicht mehr in der DOM und man hätte lange, den Fehler zu finden. Ergibt das Sinn?